app/src/org/commcare/activities/DrawActivity.java

```diff
@@ -287,18 +308,27 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
     private void createQuitDrawDialog() {
         final PaneledChoiceDialog dialog = new PaneledChoiceDialog(this, alertTitleString);
 
-        View.OnClickListener keepChangesListener = v -> saveAndClose();
-        DialogChoiceItem keepOption = new DialogChoiceItem(getString(R.string.keep_changes), -1,
-                keepChangesListener);
-
         View.OnClickListener discardChangesListener = v -> cancelAndClose();
         DialogChoiceItem discardOption = new DialogChoiceItem(getString(R.string.do_not_save), -1,
                 discardChangesListener);
 
-        dialog.setChoiceItems(new DialogChoiceItem[]{keepOption, discardOption});
+        if (isDrawn) {
+            View.OnClickListener keepChangesListener = v -> saveAndClose();
+            DialogChoiceItem keepOption = new DialogChoiceItem(getString(R.string.keep_changes), -1,
+                    keepChangesListener);
+            dialog.setChoiceItems(new DialogChoiceItem[]{keepOption, discardOption});
+        } else {
+            dialog.setChoiceItems(new DialogChoiceItem[]{discardOption});
+        }
 
         dialog.addButton(getString(R.string.cancel), v -> dialog.dismiss());
 
         dialog.showNonPersistentDialog();
     }
+
+    @Override
+    public void drawn() {
+        isDrawn = true;
```

Comment:
I feel that this variable is redundant. It's value is the same as the enabled state of `saveAndCloseButton`. 
I think instead of `if (isDrawn) {` check we can replace it with `if (saveAndCloseButton.isEnabled())` check and avoid the need of an extra variable. 
@shubham1g5 Thoughts?

Reply:
that sounds right to me.